plugins/modules/smart_proxy.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: smart_proxy
+version_added: 1.4.0
+short_description: Manage Smart Proxies
+description:
+  - Create, update and delete Smart Proxies
+author:
+  - "James Stuart (@jstuart)"
+  - "Matthias M Dellweg (@mdellweg)"
+options:
+  name:
+    description:
+      - Name or title of the Smart Proxy
+    required: true
+  url:
+    description:
+      - URL of the Smart Proxy
+    required: true
+  download_policy:
+    description:
+      - The download policy for the Smart Proxy
+      - Only used for Katello installations.
+    choices:
+      - background
+      - immediate
+      - on_demand
+    default: on_demand
+    required: false
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state
+  - theforeman.foreman.foreman.taxonomy
+'''
+
+EXAMPLES = '''
+# Create a local Smart Proxy
```

Comment:
With Katello you can't create proxies like that (they need certs and stuff), so the examples and docs should probably reflect that oddity.

Reply:
I added a katello version of the test. Nothing special was needed. Can you elaborate?

Reply:
Yeah, in tests that'll work, but the proxy won't work in a real world scenario as the proxy won't be set up properly server side

Reply:
Is that preparation, postparation or missing features of this very module?

In the former two cases, i'd rather leave that as an exercise to the interested role writer.

Reply:
This very module won't be able to create a working Katello proxy on it's own (but can be certainly used to mimic parts ot the foreman-proxy-content scenario of the foreman installer using Ansible).

I think this module should just gain a notes section in the docs pointing out that Katello proxies should usually be created using the installer.

We could also warn/fail if users try to, but I don't think that's needed.

Reply:
I added a note about using `foreman-installer` to create an actual smartproxy.
And imho, it is not even katello specific.

Reply:
It is (there is no foreman-proxy scenario in the installer, only foreman-proxy-content ;))

Thanks!